tests/testthat/test-weighting.R

```diff
@@ -34,11 +34,48 @@ with_mock_crunch({
 
     test_that("weightVariables method", {
         expect_identical(weightVariables(newds), "birthyr")
-        with(temp.option(crunch.namekey.dataset="name"), {
+        with(temp.option(crunch.namekey.dataset ="name"), {
             expect_identical(weightVariables(newds), "Birth Year")
         })
         expect_identical(weightVariables(oldds), c())
     })
+
+    test_that("generateWeightEntry errors correctly", {
+        expect_error(generateWeightEntry("bad_formula"),
+            paste0(dQuote("bad_formula"),
+                " is not a valid formula, use the form ds$var ~ c(10, 20, 30)"), fixed = TRUE)
+        expect_error(generateWeightEntry(object),
+            paste0(dQuote("object"),
+                " is not a valid formula, use the form ds$var ~ c(10, 20, 30)"), fixed = TRUE)
+        expect_error(generateWeightEntry(oldds$birthyr ~ c(30, 30, 40)),
+            "oldds$birthyr is not a categorical crunch variable", fixed = TRUE)
+        expect_error(generateWeightEntry(oldds$gender ~ c(10, 10, 10, 10, 10, 50)),
+            "Number of targets does not match number of categories for oldds$gender", fixed = TRUE)
+        expect_error(generateWeightEntry(oldds$gender ~ c(30, 20, 30)),
+            "Targets do not add up to 100% for oldds$gender", fixed = TRUE)
+        expect_error(generateWeightEntry(oldds$gender ~ c("a", "b", "c")),
+            "Targets are not numeric for oldds$gender", fixed = TRUE)
+    })
+    expected_weight_definition <- list(
+        name = "weight",
+        derivation = list(`function` = "rake",
+            args = list(
+                list(variable = "https://app.crunch.io/api/datasets/1/variables/gender/",
+                    targets = list(c(1, 0.2), c(2, 0.3), c(3, 0.5)
+                    )
+                )
+            )
+        )
+    )
+    test_that("makeWeight generates the expected VariableDefinition", {
```

Comment:
Could you break these tests into separate `test_that`s that have more informative descriptions? would make it easier to comprehend what exactly the test is asserting (and thus easier to interpret should it fail).